packages/extension/src/main.ts

```diff
@@ -90,7 +90,13 @@ function fmtDocument(document: vscode.TextDocument) {
 		if (!editor) return; // Need an editor to apply text edits.
 
 		const contents = editor.document.getText();
-		const formatted = dioxus.format_file(contents, !editor.options.insertSpaces, editor.options.tabSize);
+		let tabSize: number;
+		if (typeof editor.options.tabSize === 'number') {
```

Comment:
The docs say this is always a number when being read.

Reply:
I'm not sure about the docs, but when I tried to compile this the type of `editor.options.insertSpaces` is `number | string`. It could be something with my setup. I might have an older version of the type definitions. If you can get the extension compiling on your machine without this check, let me know.

If you are interested in testing this, you can follow the build settings [here](https://code.visualstudio.com/api/working-with-extensions/bundling-extension). Dioxus has the webpack config already setup

Reply:
Yeah, it definitely needs changing to make the types line up. I'm saying at runtime,  you'll always get back a number, so the if branch will always match (though it won't hurt).

It's typed as `number | string` because you can assign `"auto"` to it to have it to `indentSize`, but when you read it, it will always be a `number`.